ocpmodels/trainers/base_trainer.py

```diff
@@ -393,7 +394,12 @@ def _forward(self, batch):
 
         # enable gradient wrt input.
         if "grad_input" in self.config["task"]:
-            batch.x = batch.x.requires_grad_(True)
+            if self.config["model"] == "schnet":
+                inp_for_grad = batch.pos
+                batch.pos = batch.pos.requires_grad_(True)
```

Comment:
We can probably use this for all the models since only interested in the gradients wrt to positions? But could be some compatibility things to change, fine for now.